src/compiler/checker.ts

```diff
@@ -45131,37 +45133,45 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         }
     }
 
-    function checkAssertClause(declaration: ImportDeclaration | ExportDeclaration) {
-        if (declaration.assertClause) {
-            const validForTypeAssertions = isExclusivelyTypeOnlyImportOrExport(declaration);
-            const override = getResolutionModeOverrideForClause(declaration.assertClause, validForTypeAssertions ? grammarErrorOnNode : undefined);
-            if (validForTypeAssertions && override) {
-                if (!isNightly()) {
-                    grammarErrorOnNode(declaration.assertClause, Diagnostics.resolution_mode_assertions_are_unstable_Use_nightly_TypeScript_to_silence_this_error_Try_updating_with_npm_install_D_typescript_next);
-                }
-
+    function checkImportAttributes(declaration: ImportDeclaration | ExportDeclaration) {
+        if (declaration.attributes && declaration.attributes.token === SyntaxKind.AssertKeyword) {
+            grammarErrorOnFirstToken(declaration.attributes, Diagnostics.The_assert_keyword_in_import_attributes_is_deprecated_and_has_been_replaced_by_the_with_keyword);
+        }

```

Comment:
I think we need to be a little gentler on this transition. We need at least 1-2 releases where we accept both gracefully so that users can migrate a file at a time. Another strategy would be to mark it as deprecated in 5.5 and remove it entirely in 6.0.

I would be okay with being aggressive and allowing both in 5.3, and erroring in 5.4.


```suggestion
```

@RyanCavanaugh might have thoughts on this.

Reply:
It sounds like either way we want to remove the error in 5.3, right?